File: app/src/main/java/com/philbas/demo/entity/DictionaryEntry.java

package com.philbas.demo.entity;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class DictionaryEntry {

    @SerializedName("definition")
    @Expose
    private String definition;
    @SerializedName("permalink")
    @Expose
    private String permalink;
    @SerializedName("thumbs_up")
    @Expose
    private Integer thumbsUp;
    @SerializedName("sound_urls")
    @Expose
    private java.util.List<String> soundUrls = null;
    @SerializedName("author")
    @Expose
    private String author;
    @SerializedName("word")
    @Expose
    private String word;
    @SerializedName("defid")
    @Expose
    private Integer defid;
    @SerializedName("current_vote")
    @Expose
    private String currentVote;
    @SerializedName("written_on")
    @Expose
    private String writtenOn;
    @SerializedName("example")
    @Expose
    private String example;
    @SerializedName("thumbs_down")
    @Expose
    private Integer thumbsDown;

    public String getDefinition() {
        return definition;
    }

    public void setDefinition(String definition) {
        this.definition = definition;
    }

    public String getPermalink() {
        return permalink;
    }

    public void setPermalink(String permalink) {
        this.permalink = permalink;
    }

    public Integer getThumbsUp() {
        return thumbsUp;
    }

    public void setThumbsUp(Integer thumbsUp) {
        this.thumbsUp = thumbsUp;
    }

    public java.util.List<String> getSoundUrls() {
        return soundUrls;
    }

    public void setSoundUrls(java.util.List<String> soundUrls) {
        this.soundUrls = soundUrls;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getWord() {
        return word;
    }

    public void setWord(String word) {
        this.word = word;
    }

    public Integer getDefid() {
        return defid;
    }

    public void setDefid(Integer defid) {
        this.defid = defid;
    }

    public String getCurrentVote() {
        return currentVote;
    }

    public void setCurrentVote(String currentVote) {
        this.currentVote = currentVote;
    }

    public String getWrittenOn() {
        return writtenOn;
    }

    public void setWrittenOn(String writtenOn) {
        this.writtenOn = writtenOn;
    }

    public String getExample() {
        return example;
    }

    public void setExample(String example) {
        this.example = example;
    }

    public Integer getThumbsDown() {
        return thumbsDown;
    }

    public void setThumbsDown(Integer thumbsDown) {
        this.thumbsDown = thumbsDown;
    }

}

File: app/src/main/java/com/philbas/demo/entity/DictionaryEntry.kt
package com.philbas.demo.entity

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class DictionaryEntry {

    @SerializedName("definition")
    @Expose
    var definition: String? = null
    @SerializedName("permalink")
    @Expose
    var permalink: String? = null
    @SerializedName("thumbs_up")
    @Expose
    var thumbsUp: Int? = null
    @SerializedName("sound_urls")
    @Expose
    var soundUrls: List<String>? = null
    @SerializedName("author")
    @Expose
    var author: String? = null
    @SerializedName("word")
    @Expose
    var word: String? = null
    @SerializedName("defid")
    @Expose
    var defid: Int? = null
    @SerializedName("current_vote")
    @Expose
    var currentVote: String? = null
    @SerializedName("written_on")
    @Expose
    var writtenOn: String? = null
    @SerializedName("example")
    @Expose
    var example: String? = null
    @SerializedName("thumbs_down")
    @Expose
    var thumbsDown: Int? = null

}
